src/Shipping/SyncerHooks.php

```diff
@@ -0,0 +1,131 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Shipping;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Settings as GoogleSettings;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\JobRepository;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\UpdateShippingSettings;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantCenterService;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class SyncerHooks
+ *
+ * Hooks to various WooCommerce and WordPress actions to automatically sync shipping settings.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Shipping
+ *
+ * @since x.x.x
+ */
+class SyncerHooks implements Service, Registerable {
+
+	/**
+	 * This property is used to avoid scheduling duplicate jobs in the same request.
+	 *
+	 * @var bool
+	 */
+	protected $already_scheduled = false;
+
+	/**
+	 * @var GoogleSettings
+	 */
+	protected $google_settings;
+
+	/**
+	 * @var MerchantCenterService
+	 */
+	protected $merchant_center;
+
+	/**
+	 * @var UpdateShippingSettings
+	 */
+	protected $update_shipping_job;
+
+	/**
+	 * SyncerHooks constructor.
+	 *
+	 * @param MerchantCenterService $merchant_center
+	 * @param GoogleSettings        $google_settings
+	 * @param JobRepository         $job_repository
+	 */
+	public function __construct( MerchantCenterService $merchant_center, GoogleSettings $google_settings, JobRepository $job_repository ) {
+		$this->google_settings     = $google_settings;
+		$this->merchant_center     = $merchant_center;
+		$this->update_shipping_job = $job_repository->get( UpdateShippingSettings::class );
+	}
+
+	/**
+	 * Register the service.
+	 */
+	public function register(): void {
+		// only register the hooks if Merchant Center account is connected and the user has chosen for the shipping rates to be synced from WooCommerce settings.
+		if ( ! $this->merchant_center->is_connected() || ! $this->google_settings->should_get_shipping_rates_from_woocommerce() ) {
+			return;
+		}
+
+		$update_settings = function () {
+			$this->handle_update_shipping_settings();
+		};
+
+		// After a shipping zone object is saved to database.
+		add_action( 'woocommerce_after_shipping_zone_object_save', $update_settings, 90 );
```

Comment:
❔ Just a question for a WP/WC newbie like me, how could we find this hook name? I couldn't find it from [WooCommerce Action and Filter Hook Reference](https://woocommerce.github.io/code-reference/hooks/hooks.html) 🤔.